### ISSUE TITLE
make a course pack of assignments and allow all instructors to copy from some exemplar course

### ISSUE THREAD

**Issue by bnmnetp** [RunestoneInteractive/rs#261]
I'm not sure how we would handle dates?

**Comment by Tanaquil18** [comment]
It would be good to have a way for the instructor to bulk adjust dates and times.  A spreadsheet-type view of assignments and the ability to set all assignments to be due at 10am, for example.  

xyzhomework is the exemplar for this, as far as I'm aware.  You can see all assignments in a spreadsheet-type view and if you change the date of one assignment, you can hit "send down the line" and it will adjust all later assignments by the same amount, while considering weekdays only.

**Comment by bnmnetp** [comment]
Instructors can now copy assignments from the base course

**Comment by bnmnetp** [comment]
If the person who is making the assignments for the course pack is aware of the term_start_date for the base course then they can just create the assignments for the course pack as if they were in that timeline.  Then the deltas will work out fine.

Making a spreadsheet like interface to adjust all dates is a good idea for a future project.  But not until we get off of the web2py framework.

**Comment by bhoffman0** [comment]
@bnmnetp I created the csawesome assignments and tried copy all. Is it possible to fix the issues below?
1. Is it possible block the timed assessments built into the book from being copied in when you choose copy all? These were all copied in when I tried it (see ap_bowl on below). I cannot rename them, and they are in the wrong order, and some are just for PD and not relevant to students.
<img width="231" alt="image" src="https://github.com/RunestoneInteractive/rs/assets/4571962/2e606696-9af6-4387-983c-b5f41bf8caf6">

2. Can you also copy in the **Show as Timed Assessment** setting? It is not copied in currently and teachers will forget to do this.

3. Can you also copy in the **Peer Instruction Activity** setting?  It is not copied in currently.

Thanks,
Beryl

**Comment by bnmnetp** [comment]
OK,  it was easier than I thought.  Someone had cleverly added a flag to the assignments called `from_source` so we would know if an assignment came from the text.  🧐 

It was easy to add the other flags (which did not exist back when we added the copy feature) to the copy function.  It will alll be in a release that I will push out hopefully tomorrow.

**Comment by bhoffman0** [comment]
Thank you!

**Comment by seggers** [comment]
Where do I find the course pack for the fopp course for CSAwesome? This is my first python course but have built many Java and JavaScript courses. Thank you

**Comment by bnmnetp** [comment]
The question that was asked in the google group was about trying to make one.  I would love to see a small group of people who have taught from fopp come together and create a set of assignments.  I would be happy to accept that as a PR so it could be part of the book.

I simply don't have time to take on this project.

**Comment by bnmnetp** [comment]
Closing — the concrete asks here were done. Instructors can copy assignments from the base course, and the flags @bhoffman0 identified as missing were added to the copy: `from_source` assignments (the timed assessments built into the book) are excluded from copy-all, and the **Show as Timed Assessment** and **Peer Instruction Activity** settings now come along with a copied assignment.

Two things from the thread that weren't done, both better as their own issues if still wanted:

- A spreadsheet-style view for bulk date adjustment — @Tanaquil18's "send down the line" idea from xyzhomework. That was deferred until we were off web2py, and the assignment builder has since been rewritten, so it could be reconsidered against the new UI.
- An actual FOPP course pack. @seggers, that one is a call for volunteers rather than a code change; a set of assignments contributed as a PR would be accepted into the book.
